src/oss/langchain-test.mdx

```diff
@@ -1,3 +1,1057 @@
+
+:::python
+```python
+from itertools import repeat
+from langchain_core.language_models.fake_chat_models import GenericFakeChatModel
+from langchain.agents import create_agent, AgentState
+from langchain_core.messages import AIMessage
+from langgraph.runtime import Runtime
+from langgraph.checkpoint.memory import InMemorySaver
+
+small_model = GenericFakeChatModel(messages=repeat("small response", 3))
+large_model = GenericFakeChatModel(messages=repeat("large response", 1))
+
+def choose_model(state: AgentState, runtime: Runtime):
+    """Choose model based on conversation complexity."""
+    messages = state["messages"]
+    ai_message_count = sum(isinstance(m, AIMessage) for m in messages)
+    
+    return large_model if ai_message_count > 2 else small_model
+
+agent = create_agent(
+    choose_model, 
+    tools=[],
+    checkpointer=InMemorySaver()
+)
+
+def test_model_switching():
+    # Simulate 4 conversation turns
+    for i in range(4):
+        result = agent.invoke(
+            {"messages": [{"role": "user", "content": f"message {i}"}]}, 
+            {"configurable": {"thread_id": "1"}}
+        )
+    
+    assert result["messages"][-1].content_blocks == [{'type': 'text', 'text': 'large response'}]
+```
+:::
+
+:::js
+```typescript
+import { FakeChatModel } from "@langchain/core/utils/testing";
+import { createAgent } from "langchain"
+import { AIMessage } from "@langchain/core/messages";
+import { MemorySaver } from "@langchain/langgraph";
+
+const smallModel = new FakeChatModel({
+  responses: Array(3).fill("small response")
+});
+
+const largeModel = new FakeChatModel({
+  responses: ["large response"]
+});
+
+function chooseModel(state: any): any {
+  const messages = state.messages;
+  const aiMessageCount = messages.filter((m: any) => m instanceof AIMessage).length;
+
+  console.log(messages);
+  console.log(aiMessageCount);
+
+  return aiMessageCount > 2 ? largeModel : smallModel;
+}
+
+const agent = createAgent({
+  llm: chooseModel,
+  tools: [],
+  checkpointer: new MemorySaver()
+});
+
+async function testModelSwitching() {
+  // Simulate 4 conversation turns
+  for (let i = 0; i < 4; i++) {
+    const result = await agent.invoke(
+      { messages: [{ role: "user", content: `message ${i}` }] },
+      { configurable: { thread_id: "1" } }
+    );
+  }
+  
+  expect(result.messages[result.messages.length - 1].contentBlocks).toBe([{type: "text", text: "large response"}]);
+}
+```
+:::
+
+## Integration testing
+
+Many agent behaviors only emerge when using a real LLM, such as which tool the agent decides to call, how it formats responses, or whether a prompt modification affects the entire execution trajectory. LangChain offers an [`agentevals`](https://github.com/langchain-ai/agentevals) package that provides evaluators specifically designed for testing agent trajectories with live models.
```

Comment:
It could be helpful to define an agent trajectory a bit more explicitly.

For example,

> An agent trajectory is a sequence of decisions that the agent takes (e.g., which tool to call, how to format a response) etc.

The content is already in the paragraph, I think a re-arrangement to make it more explicit what a trajectory is could be helpful